guides/kontrol-example/linked-library-example.md

```diff
@@ -0,0 +1,165 @@
+
+``` sol
+vm.assume(msg.sender != address(64532647442654720331341006915390590852536243434));
+vm.assume(msg.sender != address(728815563385977040452943777879061427756277306518));
+vm.assume(msg.sender != address(856800046449367729202688009541038160569764461356));
+```
+
+...or we can assume that `CALLER_ID` is equals to a concrete address.
+
+```sol
+vm.assume(msg.sender == address(0xc0ffee));
+```
+
+Let's go with the first approach, updating the test function to:
+
+```sol
+function testSquare(uint256 n) public {
+    vm.assume(msg.sender != address(64532647442654720331341006915390590852536243434));
+    vm.assume(msg.sender != address(728815563385977040452943777879061427756277306518));
+    vm.assume(msg.sender != address(856800046449367729202688009541038160569764461356));
+    uint256 result = SimpleMath.square(n);
+    assertEq(result, n * n);
+}
+```
+
+Next, we must rebuild the project with `kontrol build --regen --rekompile`.
+With this approach, we preserve the previous proof executions.
+Now let's run the new test with `kontrol prove --mt LinkedLibTest.testSquare --reinit`.
+
+The proof still fails, but we see only one pending node and one failing node.
+There is one more branch that takes place.
+
+We could inspect the kcfg once more to see what the control flow graph looks like now, but that is optional, as the information we need right now is also present in the final output of Kontrol.
+
+<figure><img src="../../.gitbook/assets/linked_libs_5.png" alt=""><figcaption><p>Kontrol_TUI_2</p></figcaption></figure>
+
+The output summarizes the first failure identified while running the proof.
+We determine if a node represents a failure based on multiple conditions, such as a revert status code or a failed assertion.
+[You can learn more about the success predicate here](https://github.com/runtimeverification/kontrol/blob/master/src/kontrol/kdist/foundry.md#foundry-success-predicate).
+It has three components: the failure reason, the path condition, and the model.
+The output shows that the status code is EVMC_REVERT.
+
+Let's focus on the path condition for a bit.
+We can see that it is a conjunction of two negated conditions:
+    1. `#Not ( { VV0_n_114b9705:Int #Equals chop ( VV0_n_114b9705:Int *Int VV0_n_114b9705:Int ) /Word VV0_n_114b9705:Int } )`
+    2. `#Not ( { VV0_n_114b9705:Int #Equals 0 } )`
+
+The second one is pretty straightforward, asserting that the symbolic argument `VV0_n_114b9705`, is non-zero.
+For the first one, we can simplify the variable name to obtain `#Not( n #Equals chop (n *Int n /Word n))`.
+Let's break it apart:
+- `chop()` is an internal function that ensures that it's argument is lower than `2^256`.
+Think about it as a (`_%_`) mod operation.
+- In the K language, operators also take the sort on which they are applied.
+  `*Int` represents the product of two unbounded integers.
+
+The equation above is equivalent to `( n != (n * n / n) mod type(uint256).max)` which looks more like an overflow check.
```

Comment:
I think it might actually be 
```
( n != (n * n / n) mod (type(uint256).max + 1))
```
Because `type(uint256).max` is `2^256 - 1` (115792089237316195423570985008687907853269984665640564039457584007913129639935), and `chop` is `modInt pow256`.

Reply:
Good catch!